src/main/java/taxi/web/filter/AuthenticationFilter.java

```diff
@@ -0,0 +1,40 @@
+package taxi.web.filter;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+public class AuthenticationFilter implements Filter {
+    private Set<String> allowedUrl = new HashSet<>();
+
+    @Override
+    public void init(FilterConfig filterConfig) {
+        allowedUrl.add("/login");
+        allowedUrl.add("/registration");
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest,
+                         ServletResponse servletResponse, FilterChain filterChain)
+            throws IOException, ServletException {
+        HttpServletRequest httpServletRequest = (HttpServletRequest) servletRequest;
+        HttpServletResponse httpServletResponse = (HttpServletResponse) servletResponse;
+        HttpSession session = httpServletRequest.getSession();
+        Long driverId = (Long) session.getAttribute("driver_id");
+
+        if (driverId == null && allowedUrl.contains(httpServletRequest.getServletPath())) {
+            filterChain.doFilter(servletRequest, servletResponse);
+            return;
+        }
+        filterChain.doFilter(servletRequest, servletResponse);
```

Comment:
There is no sense in this code right now:
The condition should check that driverId is not null or the URL is allowed. And redirect to login page or perform doFilter() method.